modules/capi-yaml-machine-set-azure.adoc

```diff
@@ -0,0 +1,44 @@
+// Module included in the following assemblies:
+//
+// * machine_management/cluster_api_machine_management/cluster_api_provider_configurations/cluster-api-config-options-azure.adoc
+
+:_mod-docs-content-type: REFERENCE
+[id="capi-yaml-machine-set-azure_{context}"]
+= Sample YAML for a Cluster API compute machine set resource on {azure-full}
+
+The compute machine set resource defines additional properties of the machines that it creates.
+The compute machine set also references the infrastructure resource and machine template when creating machines.
```

Comment:
It references the Cluster and Machine template. Maybe confusion because the machine template is specified as infrastructureRef?

Reply:
Ok so I think my understanding was that it was referencing the machine template (which we are in agreement about) and what the docs have as the "infrastructure resource". I _think_ that's what you have referred to as the InfraCluster resource.

So IIUC, I have:
1. cluster
2. infrastructure
3. machine template
4. machine set (which references 2&3)

I think you are saying:
1. cluster
2. infracluster
3. machine template
4. machine set (which references 1&3)

Does that seem right?

Reply:
Yes, that's what I meant. 

There is a CRD called Infrastructure that has a single instance named cluster that has a bunch of information about the cluster. https://github.com/openshift/api/blob/master/config/v1/types_infrastructure.go
It has nothing to do with CAPI.